Adopt nan v2

bpasero opened the pull request:
I can compile against Electron 0.32.0 with the following command:

HOME=~/.electron-gyp node-gyp rebuild --target=0.32.0 --arch=x64 --dist-url=https://atom.io/download/atom-shell

I am not 100% sure about the change for Nan::HandleScope();

paulcbetts commented:
@bpasero HandleScope changes look good

bpasero commented:
@paulcbetts ok. is there a way to tell travis to compile against io.js 3.x?

zcbenz commented:
@bpasero You can follow this:
https://github.com/atom/node-runas/blob/master/.travis.yml

zcbenz commented on src/onig-scanner-worker.cc:
```diff
@@ -15,17 +15,17 @@ void OnigScannerWorker::Execute() {
 
 void OnigScannerWorker::HandleOKCallback() {
-  NanScope();
+  Nan::HandleScope();
```
We need to give it a name since `Nan::HandleScope` is a class and no longer a macro:

```c++
Nan::HandleScope scope;
```

bpasero commented:
@zcbenz thanks, fixed. 

I changed the travis build to use Linux with g++ 4.8 and now it runs through. Note that I rely on the tests to be green, I did not test node-oniguruma in the product.

zcbenz commented:
Thanks!

bpasero commented:
Nice. Once a new release is out @ node-oniguruma, I can verify in our build. Thanks!

bpasero commented:
Ah lol, disregard, just noticed the 5.0.0 release. Cool :)
